Reachable Assertion Panic in `timespec_t::from()` by negative `tv_sec` value

### Describe the bug
There is a reachable assertion in `timespec_t::from()` at kernel/src/time/mod.rs:50 when call syscall `utimes` with negative `tv_sec` value.

https://github.com/asterinas/asterinas/blob/2bda8d17c42652c6c9276a83e88600567b55fea6/kernel/src/time/mod.rs#L46-L53

### To Reproduce
1. Compile a program which calls `utimes`: 
```C
#define _GNU_SOURCE
#include <stdio.h>
#include <sys/time.h>
#include <sys/syscall.h>
#include <unistd.h>
#include <errno.h>

int main(int argc, char *argv[]) {
  struct timeval times[2];
  times[0].tv_sec = -1;
  times[0].tv_usec = 10;
  times[1].tv_sec = -1;
  times[1].tv_usec = 10;

  syscall(SYS_utimes, "/dev/null", times);
  perror("utimes");

  return 0;
}
```
2. Run the compiled program in Asterinas.

### Expected behavior

Asterinas reports panic and is terminated.

### Environment
- Official docker asterinas/asterinas:0.8.1
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 2bda8d1

### Logs
```
~ # /root/utimes.c 
panicked at /root/asterinas/kernel/src/time/mod.rs:50:9:
assertion failed: sec >= 0
Printing stack trace:
   1: fn 0xffffffff88860d60 - pc 0xffffffff88860dca / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b029770;

   2: fn 0xffffffff88860b70 - pc 0xffffffff88860cc0 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b0297d0;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b029920;

   4: fn 0xffffffff889d2bf0 - pc 0xffffffff889d2c72 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b029930;

   5: fn 0xffffffff889d2d90 - pc 0xffffffff889d2dd0 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b0299c0;

   6: fn 0xffffffff88344840 - pc 0xffffffff883448a6 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b029a30;

   7: fn 0xffffffff8859d700 - pc 0xffffffff8859d9ab / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b029a70;

   8: fn 0xffffffff8859bd60 - pc 0xffffffff8859bdc4 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b029c90;

   9: fn 0xffffffff885507b0 - pc 0xffffffff8857df4d / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b029f70;

  10: fn 0xffffffff883f79e0 - pc 0xffffffff883f7a6e / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b040410;

  11: fn 0xffffffff8848e0d0 - pc 0xffffffff8848ec76 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b0405b0;

  12: fn 0xffffffff8820f570 - pc 0xffffffff8820f57e / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b040f90;

  13: fn 0xffffffff88898970 - pc 0xffffffff88898986 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b040fb0;

  14: fn 0xffffffff88888810 - pc 0xffffffff88888879 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b040fd0;


     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff8883b1a7; rbp                0x0; rsp 0xffff80000b041000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:113
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) cbtrlxuzeok8ueszfa474zvwu:?
(  4) ??:?
(  5) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:220
(  6) ??:?
(  7) /root/asterinas/kernel/src/syscall/utimens.rs:195
(  8) /root/asterinas/kernel/src/syscall/utimens.rs:71
(  9) /root/asterinas/kernel/src/syscall/mod.rs:164
( 10) /root/asterinas/kernel/src/syscall/mod.rs:328
( 11) /root/asterinas/kernel/src/thread/task.rs:69
( 12) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 13) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 14) /root/asterinas/ostd/src/task/mod.rs:175
make: *** [Makefile:173: run] Error 1
```

--- a/kernel/src/syscall/utimens.rs
+++ b/kernel/src/syscall/utimens.rs
@@ -189,7 +189,13 @@ fn do_futimesat(
 ) -> Result<SyscallReturn> {
     let times = if timeval_ptr != 0 {
         let (autime, mutime) = read_time_from_user::<timeval_t>(timeval_ptr, ctx)?;
-        if autime.usec >= 1000000 || autime.usec < 0 || mutime.usec >= 1000000 || mutime.usec < 0 {
+        if autime.usec >= 1000000
+            || autime.usec < 0
+            || autime.sec < 0
+            || mutime.usec >= 1000000
+            || mutime.usec < 0
+            || mutime.sec < 0
+        {
             return_errno_with_message!(Errno::EINVAL, "Invalid time");
         }
         let (autime, mutime) = (timespec_t::from(autime), timespec_t::from(mutime));

